internal/exec/aws_eks_update_kubeconfig.go

```diff
@@ -0,0 +1,215 @@
+	clusterName := kubeconfigContext.ClusterName
+
+	// Optionally specify a kubeconfig file to append with your configuration. By default, the configuration is written to the first file path
+	// in the KUBECONFIG environment variable (if it is set) or the default kubeconfig path (.kube/config) in your home directory
+	kubeconfigPath := kubeconfigContext.Kubeconfig
+
+	// Alias for the cluster context name. Defaults to match cluster ARN
+	alias := kubeconfigContext.Alias
+
+	// Check if all the required parameters are provided to execute the command without needing `atmos.yaml` config and context
+	// The rest of the parameters are optional
+	requiredParamsProvided := clusterName != "" && (profile != "" || roleArn != "")
+
+	shellCommandWorkingDir := ""
+
+	if !requiredParamsProvided {
+		// If stack is not provided, calculate the stack name from the context (tenant, environment, stage)
+		if kubeconfigContext.Stack == "" {
+			err := c.InitConfig()
+			if err != nil {
+				return err
+			}
+
+			if len(c.Config.Stacks.NamePattern) < 1 {
+				return errors.New("stack name pattern must be provided in 'stacks.name_pattern' CLI config or 'ATMOS_STACKS_NAME_PATTERN' ENV variable")
+			}
+
+			stack, err := c.GetStackNameFromContextAndStackNamePattern(kubeconfigContext.Tenant,
+				kubeconfigContext.Environment, kubeconfigContext.Stage, c.Config.Stacks.NamePattern)
+			if err != nil {
+				return err
+			}
+
+			kubeconfigContext.Stack = stack
+		}
+
+		var configAndStacksInfo c.ConfigAndStacksInfo
+		configAndStacksInfo.ComponentFromArg = kubeconfigContext.Component
+		configAndStacksInfo.Stack = kubeconfigContext.Stack
+
+		configAndStacksInfo.ComponentType = "terraform"
+		configAndStacksInfo, err := ProcessStacks(configAndStacksInfo, true)
+		shellCommandWorkingDir = path.Join(c.ProcessedConfig.TerraformDirAbsolutePath, configAndStacksInfo.ComponentFolderPrefix, configAndStacksInfo.FinalComponent)
+		if err != nil {
+			configAndStacksInfo.ComponentType = "helmfile"
+			configAndStacksInfo, err = ProcessStacks(configAndStacksInfo, true)
+			shellCommandWorkingDir = path.Join(c.ProcessedConfig.HelmfileDirAbsolutePath, configAndStacksInfo.ComponentFolderPrefix, configAndStacksInfo.FinalComponent)
+			if err != nil {
+				return err
+			}
+		}
+
+		context := c.GetContextFromVars(configAndStacksInfo.ComponentVarsSection)
+
+		// `kubeconfig` can be overridden on the command line
+		if kubeconfigPath == "" {
+			kubeconfigPath = fmt.Sprintf("%s/%s-kubecfg", c.Config.Components.Helmfile.KubeconfigPath, kubeconfigContext.Stack)
+		}
+		// `clusterName` can be overridden on the command line
+		if clusterName == "" {
+			clusterName = c.ReplaceContextTokens(context, c.Config.Components.Helmfile.ClusterNamePattern)
+		}
+		// `profile` can be overridden on the command line
+		if profile == "" {
+			profile = c.ReplaceContextTokens(context, c.Config.Components.Helmfile.HelmAwsProfilePattern)
+		}
```

Comment:
Avoid any possible errors trying to derive a profile name if the profile is not going to be used anyway. 

```suggestion
		if profile == "" && roleArn == "" {
			profile = c.ReplaceContextTokens(context, c.Config.Components.Helmfile.HelmAwsProfilePattern)
		}
```